contracts/proxy/src/contract.rs

```diff
@@ -404,6 +406,25 @@ pub fn execute_update_guardians(
     }
 }
 
+/// Update label by user
+pub fn execute_update_label(
+    deps: DepsMut,
+    info: MessageInfo,
+    env: Env,
+    new_label: String,
+) -> Result<Response, ContractError> {
+    let is_user = ensure_is_user(deps.as_ref(), info.sender.as_str());
+    let is_contract = ensure_is_contract_self(&env, &info.sender);
+    if is_user.is_err() && is_contract.is_err() {
+        is_user?;
+        is_contract?;
+    }
+    LABEL.save(deps.storage, &new_label)?;
```

Comment:
I know this a bit silly but we may check if new label == old label to avoid unnecessary cost.

Reply:
I agree